### ISSUE TITLE
A question about Five-fold Cross-validation

### ISSUE THREAD

**Issue by TRRtrr** [yxonic/DTransformer#4]
**Hello**, I noticed while running this code that each dataset has only one training set and one testing set. However, it is mentioned in this paper that the final answer was obtained using a Five-fold Cross-validation approach. I would like to ask, do I need to manually cut out a validation set from the training set? Looking forward to receiving your guidance very much, thank you very much!

**Comment by yxonic** [comment]
The cross-validation experiments were done manually in jupyter notebooks, and I didn't have time to clean up the messy code there. So you may have to do this yourself, unfortunately.

**Comment by TRRtrr** [comment]
Hello, sorry to bother you again. As I am not familiar with the 5-fold cross-validation technology in the KT field, I would like to ask you another question. Which of the following methods is commonly used for 5-fold cross-validation in the KT field?

1) Manually cut out 5 training sets, 5 validation sets, and 5 testing sets;

2) Manually cut out 5 training sets and 5 validation sets, and then share 1 test set;

3) Write a piece of code that can automatically extract 5 training sets, 5 validation sets, and 5 test sets randomly from the original dataset.

I look forward to your guidance very much, and I would greatly appreciate it. Thank you!

**Comment by yxonic** [comment]
> 2\. Manually cut out 5 training sets and 5 validation sets, and then share 1 test set;

In our experiment, we use this method, and cut training/validation set based on student. That is, for each validation step, 1/5 of the students are placed in the validation set. We use cross-validation only for hyperparameter tuning, and the test set is the same for all the experiments.

**Comment by TRRtrr** [comment]
Thank you very much for your answer. It has been particularly helpful to me！
Due to my recent learning, the combination of function 'train_test_split' and function 'KFold' can also perform dataset partitioning for 5-fold cross-validation and the shuffle parameter in 'KFold' function can be used to reorder the dataset. Because my understanding of the impact of data on KT results is not deep enough, I am also a bit confused about the difference between manually partitioning 5-fold cross-validation dataset and using the above two functions to partition partitioning dataset. If you are familiar with this aspect, can you give me some guidance? Thank you again for your patient guidance before, thank you!

**Comment by yxonic** [comment]
I am not very familiar with KFold function, but I guess it should work!

**Comment by TRRtrr** [comment]
Thank you very much for your guidance! 
I have carefully read this paper and very interested in DTransformer model, so I have reproduced the code and found that the number of rows in my train and test dataset is slightly different from the data you provided. How can I obtain the same train and test data as your dataset？I want to see if I can reproduce the data preprocessing process by myself and I also really want to learn the data preprocessing process.
